lib/mongo/socket/ssl.rb

```diff
@@ -122,23 +124,18 @@ def create_context(options)
       end
 
       def set_cert(context, options)
-        context.cert = OpenSSL::X509::Certificate.new(File.open(options[:ssl_cert]))
+        context.cert = certificate_from_option(options[:ssl_cert])
```

Comment:
maybe we can call this ``ssl_cert_string`` or ``ssl_cert_raw`` for the pem-encoded key/cert
leave ``ssl_cert`` as meaning the file path for backwards compatibility
have ``ssl_cert_obj`` mean the OpenSSL::X509::Certificate object

and the same variations for ``ssl_ca_cert``